Source/JavaScriptCore/builtins/PromiseConstructor.js

```diff
@@ -246,9 +246,9 @@ function allSettled(iterable)
         for (var value of iterable) {
             @putByValDirect(values, index, @undefined);
             var nextPromise = promiseResolve.@call(this, value);
-            var [resolveElement, rejectElement] = newResolveRejectElements(index);
+            var resolveRejectElements = newResolveRejectElements(index);
```

Comment:
Let's return object instead of array too.

Reply:
Does your comment mean the following code?

```javascript
var resolveRejectElements = newResolveRejectElements(index);
++remainingElementsCount;
nextPromise.then(resolveRejectElements.resolve, resolveRejectElements.reject);
```

If so, I believe we should keep it as an array because this code would cause tests like [this one](https://github.com/tc39/test262/blob/main/test/built-ins/Promise/all/resolve-element-function-name.js) to fail.

Reply:
You can avoid that issue like this.
```
var object = {
    test: (void 0, () = > { })
};
object.test.name // ""
```

We should change that since the above object literal based one is much better in terms of efficacy compared to array.

Reply:
But I think another option you can take is just inline entire function, and using let scoped variables instead, probably that's better in terms of efficacy.
```
let currentIndex = index++;
let alreadyCalled = false;
nextPromise.then(
            (value) => {
                if (alreadyCalled)
                    return @undefined;
                alreadyCalled = true;
                var obj = {
                    status: "fulfilled",
                    value
                };
                @putByValDirect(values, currentIndex, obj);
                --remainingElementsCount;
                if (remainingElementsCount === 0)
                    return promiseCapability.resolve.@call(@undefined, values);
                return @undefined;
            },
            (reason) => {
                if (alreadyCalled)
                    return @undefined;
                alreadyCalled = true;
                var obj = {
                    status: "rejected",
                    reason
                };
                @putByValDirect(values, currentIndex, obj);
                --remainingElementsCount;
                if (remainingElementsCount === 0)
                    return promiseCapability.resolve.@call(@undefined, values);
                return @undefined;
            });

Reply:
Thank you. Your second option sounds good to me, I've updated the patch.